pymc/tests/test_distributions_timeseries.py

```diff
@@ -17,15 +17,51 @@
 
 from pymc.aesaraf import floatX
 from pymc.distributions.continuous import Flat, Normal
-from pymc.distributions.timeseries import AR, AR1, GARCH11, EulerMaruyama
+from pymc.distributions.timeseries import (
+    AR,
+    AR1,
+    GARCH11,
+    EulerMaruyama,
+    gaussianrandomwalk,
+)
 from pymc.model import Model
 from pymc.sampling import sample, sample_posterior_predictive
 from pymc.tests.helpers import select_by_precision
 
 # pytestmark = pytest.mark.usefixtures("seeded_test")
-pytestmark = pytest.mark.xfail(reason="Timeseries not refactored")
 
 
+def test_grw_rv_op():
+    """Basic test for GRW RV op"""
+    init = 1
+    mu = 3
+    sd = 0.0000001
+    size = 4
+
+    grw = gaussianrandomwalk(mu, sd, init, size=size).eval()
+    np.testing.assert_almost_equal(grw[-1], 13, decimal=4)
+    assert grw.shape[0] == size
+
+
+def test_grw_log():
+    vals = [1, 2]
+    mu = 0
+    sd = 1
+    init = 0
+
+    import pymc as pm
+
+    from pymc.distributions.timeseries import GaussianRandomWalk
+
+    with pm.Model():
+        grw = GaussianRandomWalk("grw", mu, sd, init, size=2)
+
+    logp = pm.logp(grw, vals)
```

Comment:
Alright @ricardoV94 Im at the next stage of development challenge which is shape issue.

I get what the error is saying but I'm not sure exactly what to do about it. My questions in order is
1. is the way I'm writing the test correct e.g. Should I be testing through the pm.logp api?
2. What is the fix to the shape issue do you think?

```
 self = <aesara.tensor.basic.Join object at 0x7ff3b387ca90>, axis = 0
tens = (Elemwise{sub,no_inplace}.0, Elemwise{sub,no_inplace}.0)
as_tensor_variable_args = [Elemwise{sub,no_inplace}.0, Elemwise{sub,no_inplace}.0]
output_maker = <function Join.make_node.<locals>.output_maker at 0x7ff39e1d04d0>

    def _make_node_internal(self, axis, tens, as_tensor_variable_args, output_maker):
        if not builtins.all(targs.type.ndim for targs in as_tensor_variable_args):
            raise TypeError(
>               "Join cannot handle arguments of dimension 0."
                " For joining scalar values, see @stack"
            )
E           TypeError: Join cannot handle arguments of dimension 0. For joining scalar values, see @stack
```

https://github.com/pymc-devs/pymc/runs/4825257275?check_suite_focus=true#step:7:2040

Reply:
I replied below with a suggestion for the shape issue. The logp you should test like we do for normal distributions, creating the distribution in a model context and evaluating the model logp for a couple of points. 

I would take a look at the Multinomial tests we have in `test_distributions.py` to have an idea of the range of logp tests we have for multivariate distributions.

Also, we didn't have any previous tests for the GRW? That's surprising